dart/button_error.dart

```diff
@@ -0,0 +1,9 @@
+import 'dart:html';
+void main() {
+  for (var i = 0; i < 10; i++) {
+    var button = new ButtonElement();
+    button.text = i;
+    button.onClick.listen((e) => window.alert(i));
```

Comment:
Agreed. Then again new JavaScript doesn't either; the fat arrow provides the expected `this` context.